### PR TITLE
Make TimedeltaIndex +/- pd.NaT return TimedeltaIndex

### LINKED ISSUE #19124: TimedeltaIndex +/- NaT inconsistency with Series[timedelta64]+/- NaT
Analogous to (#18808)

Adapted from `tests.indexes.timedeltas.test_arithmetic.TestTimedeltaIndexArithmetic.test_timedelta_ops_with_missing_values`

```
tdi = pd.TimedeltaIndex(['00:00:01'])
ser = pd.Series(tdi)

>>> tdi + pd.NaT
DatetimeIndex(['NaT'], dtype='datetime64[ns]', freq=None)
>>> tdi - pd.NaT
DatetimeIndex(['NaT'], dtype='datetime64[ns]', freq=None)

>>> ser + pd.NaT
0   NaT
dtype: timedelta64[ns]
>>> ser - pd.NaT
0   NaT
dtype: timedelta64[ns]
```

i.e. `Series[timedelta64]` treats `pd.NaT` as a `timedelta`, while `TimedeltaIndex` treats `pd.NaT` as a `datetime` (in a way that makes sense for addition but not for subtraction).

Expected Behavior
Internal consistency in the sense that:
```
(tdi - pd.NaT).dtype == (ser - pd.NaT).dtype
(tdi + pd.NaT).dtype == (ser + pd.NaT).dtype
```

For `tdi - pd.NaT` it _only_ makes sense for the result to be `timedelta64`, so the `TimedeltaIndex` behavior is wrong.

For `tdi + pd.NaT` a reasonable argument can be made for the result being either `timedelta64` or `datetime64`.  Returning timedelta64 maintains consistency with `TimedeltaIndex.__sub__`, while `datetime64` maintains consistency with `TimedeltaIndex.__radd__`.  All in all I think keeping addition commutative is the more important of the two.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #19124
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Travis http error

File: pandas/tests/scalar/test_nat.py
Comment (jreback) [inline pandas/tests/scalar/test_nat.py]: can you add similar testing for Series here as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/test_nat.py
Comment (jbrockmendel) [reply pandas/tests/scalar/test_nat.py]: Shoot this surfaces a new bug; I'll xfail for now.

File: pandas/tests/scalar/test_nat.py
Comment (jreback) [inline pandas/tests/scalar/test_nat.py]: can you just parametrize around tdi and Series, too much repeated code (you can xfail that case using pytest.param)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
you need to rebase

**Comment by jbrockmendel** [comment]
Looks like Travis is timing out in a bunch of places.  Will hold off on pushing for now.  Once this gets in I've got a patch ready for #19158 that will un-xfail that one test.

**Comment by jbrockmendel** [comment]
ping.  Once this goes in I can fix the bug that surfaced in `pd.NaT - Series`

File: pandas/tests/scalar/test_nat.py
Comment (jreback) [reply pandas/tests/scalar/test_nat.py]: pls do this here.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
> can you just parametrize around tdi and Series, too much repeated code (you can xfail that case using pytest.param)

Just tried this and pytest is raising errors at collect-time, complaining about pytest.param.  This would be a non-issue if weren't for the xfail.  After this branch is merged I've got a branch ready that fixes the xfailing error.  Since that PR will have to change this test anyway, let's parametrize it in that PR.

File: pandas/tests/scalar/test_nat.py
Comment (jreback) [reply pandas/tests/scalar/test_nat.py]: you are pinging but not doing what I ask.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
Please read my previous comment.

**Comment by jbrockmendel** [comment]
> I read your comment. its doesn't make any sense.

Then ask for clarification, buddy.  Just repeating the request without acknowledging that I tried to implement and discuss it is pretty discouraging.

Implementing the parameterization you're asking for looks like this:

```
@pytest.mark.parametrize('box, assert_func', [
    (TimedeltaIndex, tm.assert_index_equal),
    pytest.param((Series, tm.assert_series_equal),
                 marks=pytest.mark.xfail(reason='NaT - Series returns NaT'))
])
def test_nat_arithmetic_td64_vector(box, assert_func):
    [...]
```

(Even if it didn't cause an error, this is 5 lines of boilerplate to eliminate a 6 line test.)

`python -m pytest pandas/tests/scalar/test_nat.py` then raises at test-collection time:

```
/usr/local/lib/python2.7/site-packages/_pytest/python.py:832: in parametrize
    param.values, argnames))
E   ValueError: In "parametrize" the number of values (((<class 'pandas.core.series.Series'>, <function assert_series_equal at 0x10c0a6140>),)) must be equal to the number of names (['box', 'assert_func'])
```

So the options here are:

- Wait for this to get fixed on pytest's end
- kludge some other boilerplate to avoid needing to use pytest.param here
- Merge as is, at which point I can make a PR that fixes the bug that makes the xfail necessary, and that PR can parametrize the test.

**Comment by jreback** [comment]
```
@pytest.mark.parametrize('box, assert_func', [
    (TimedeltaIndex, tm.assert_index_equal),
    pytest.param(Series, tm.assert_series_equal,
                 marks=pytest.mark.xfail(reason='NaT - Series returns NaT'))
])
def test_nat_arithmetic_td64_vector(box, assert_func):
    [...]
```

pass as args not a tuple. I agree its a little bit odd that pytest doesn't accept a tuple here, and I know that this is adding an xfail that you are going to remove later, but would rather do that than miss adding the tests with coverage.

**Comment by jbrockmendel** [comment]
> pass as args not a tuple.

Not sure what you mean here.

> this is adding an xfail that you are going to remove later, but would rather do that than miss adding the tests with coverage.

The xfail and the coverage exist in the PR-as-is.  If you can clarify the "args not a tuple" thing I'll try to implement it, since that is a faster way of getting bugs fixed than arguing about it.  But compromising with me on this is an even faster way to get even more bugs fixed, and it ends with the same parametrized tests you have in mind.

**Comment by jreback** [comment]
do this
```
pytest.param(Series, tm.assert_series_equal,
                 marks=pytest.mark.xfail(reason='NaT - Series returns NaT'))
```

not this
```
pytest.param((Series, tm.assert_series_equal),
                 marks=pytest.mark.xfail(reason='NaT - Series returns NaT'))
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thank you for fixing!

**Pull request merged**
